src/Core__JSON.resi

```diff
@@ -3,9 +3,17 @@ Functions for interacting with JSON.
 */
 
 /** 
-A type representing a JSON object.
+A type representing a valid JSON value.
 */
-type t = Js.Json.t
+@unboxed
+type rec t = Js.Json.t =
+  | @as(false) False
+  | @as(true) True
+  | @as(null) Null
+  | String(string)
+  | Number(float)
+  | Object(Js.Dict.t<t>)
```

Comment:
`Js.Dict.t` assumes keys are `sting`s, but they can also be `Symbol`s, according to the spec, and in practice any other primitive type as well. Although I'm not sure the latter is a significant difference since primitive values automatically coerce to and from `string`.

Reply:
Yeah, that's a good point. Currently you'll need to take care of stringifying the keys yourself.